src/aiokatcp/sensor.py

```diff
@@ -202,6 +203,9 @@ def set_value(
     ) -> None:
         """Set the current value of the sensor.
 
+        Also check if the incoming value dtype is compatible with the core dtype
+        of this sensor.
```

Comment:
```suggestion
        Also validate that the incoming value type is compatible with the type
        of this sensor.
```
"dtype" is a numpy concept, and aiokatcp doesn't use numpy.